osu.Game/Modes/UI/HitRenderer.cs

```diff
@@ -60,7 +61,12 @@ public abstract class HitRenderer<TObject> : HitRenderer
 
         protected HitRenderer(WorkingBeatmap beatmap)
         {
+            Debug.Assert(beatmap != null, "HitRenderer initialized with a null beatmap.");
+
+            // Convert + process the beatmap
             Beatmap = CreateBeatmapConverter().Convert(beatmap.Beatmap);
+            Beatmap.HitObjects.ForEach(h => CreateBeatmapProcessor().SetDefaults(h, Beatmap));

```

Comment:
- what's the point of SetDefaults when it's just doing a single operation with the two arguments provided?
- You're instantiating a BeatmapProcessor for *every* hitobject here. I'm pretty sure this isn't the intention.